azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -937,8 +963,16 @@ private String findValueForJobVariable(final String jobName, final String variab
       }
     }
 
+    // Todo jamiesjc: need to handle condition for embedded flows
     // Get job output props
-    final Props outputProps = this.getExecutableFlow().getExecutableNode(jobName).getOutputProps();
+    final ExecutableNode target = node.getParentFlow().getExecutableNode(jobName);
```

Comment:
Assume every node have a parent flow? What does the code do here?

Reply:
Yes, when an executableNode is created, azkaban.executor.ExecutableNode#setParentFlow will be called.
Instead of calling `this.getExecutableFlow()`, we should call `node.getParentFlow()` because for jobs inside an embedded flow, they should only have conditions on jobs that are also inside the same embedded flow. `this.getExecutableFlow()` would fetch the flow at root level whereas `node.getParentFlow()` would fetch the embedded flow.  Also I added a Null check for the target node. This part will be modified later to fully support condition for embedded flows.